assembly/src/ast/mod.rs

```diff
@@ -336,6 +337,26 @@ impl ProgramAst {
     pub fn clear_imports(&mut self) {
         self.import_info = None;
     }
+
+    // WRITE TO FILE
+    // --------------------------------------------------------------------------------------------
+    #[cfg(feature = "std")]
+    /// Writes ProgramAst to provided file path
+    pub const LIBRARY_EXTENSION: &'static str = "masb";
+    pub fn write_to_file<P>(&self, dir_path: P) -> io::Result<()>
+    where
+        P: AsRef<Path>,
+    {
+        fs::create_dir_all(&dir_path)?;
+        // ToDo: find out how to get the hash of the ProgramAst
+        let mut path = dir_path.as_ref().join("program_hash");
+        path.set_extension(Self::LIBRARY_EXTENSION);
```

Comment:
I think it will be too difficult to compile the file here to get program hash. Also, I'm not sure it will be very useful as it will be difficult to tell which program does what.

I think instead we should just provide a `file_path` to the function. We can also assume that the extension is a part of the file path already (so, we don't need to append it).

Reply:
So we can basically leave this as a function? 

```
    {
        fs::create_dir_all(&dir_path)?;
        let path = dir_path;

        let bytes = self.to_bytes(AstSerdeOptions {
            serialize_imports: true,
        });
        fs::write(path, bytes)
    }
```